src/raftstore/store/worker/pd.rs

```diff
@@ -174,6 +182,57 @@ impl<T: PdClient> Runner<T> {
             error!("report split failed {:?}", e);
         }
     }
+
+    fn handle_validate_peer(&self, local_region: metapb::Region, peer: metapb::Peer) {
+        metric_incr!("pd.get_region");
+        match self.pd_client.get_region(local_region.get_start_key()) {
+            Ok(mut pd_region) => {
+                metric_incr!("pd.get_region");
+                if pd_region == local_region {
+                    // Local region info is as fresh as pd, which means
+                    // local peer is still in the region, let it be there.
+                    info!("peer {:?} is still valid because local region info is as fresh as pd",
+                          peer);
+                    return;
+                }
+                if !(pd_region.get_start_key() == local_region.get_start_key()) {
```

Comment:
The example case is just one of the possible cases which could happen. Indeed the region [100, 200) of this node B could change to any value, .e.g:
```
[150, 200)
```
or

```
[120, 140)
```

Those region is between [100, 200), so the data node B holds still make sense. But it's hard to get the right `start_key` to query PD when validating the peer.

Consider this complexity and the duty of PD to manipulate and to serve query these meta data, PD should support the query of region info by region id.